src/main/scala/scala/compat/java8/StreamConverters.scala

```diff
@@ -197,19 +197,19 @@ with converterImpl.Priority1AccumulatorConverters
 
   implicit final class EnrichDoubleArraySeqWithStream(private val a: collection.mutable.ArraySeq[Double])
     extends AnyVal with MakesSequentialStream[Double, DoubleStream] with MakesParallelStream[Double, DoubleStream] {
-    def seqStream: DoubleStream = java.util.Arrays.stream(a.array)
+    def seqStream: DoubleStream = java.util.Arrays.stream(a.toArray)
```

Comment:
This operation creates a new array. OK, in fact it doesn't, but I'm pretty sure it should (https://gitter.im/scala/collection-strawman?at=5b275c382392ea56b26fe91c).

We should instead use
```
if (a.elemTag.runtimeClass == java.lang.Double.TYPE) a.array.asInstanceOf[Array[Double]] else a.toArray
```

Maybe we will put something like that (`unsafeArrayIfPossible`) directy into the `ArraySeq` class, but I think we should just write a helper method here for now.